src/Drivers/nlpSparse_ex9.hpp

```diff
@@ -0,0 +1,406 @@
+        gradf[nx_*(i+1)+j] += (x[nx_*(i+1)+j]-x[j])/double(S_);
+	gradf[j] += (x[j]-x[nx_*(i+1)+j])/double(S_);
+      }
+    }
+    //std::cout<<"gradf15 "<<gradf[15]<<std::endl;
+    return true;
+  }
+
+  /* Four constraints no matter how large n is */
+  // This constraint attempts to use existing functions
+  bool eval_cons(const long long& n, const long long& m,
+		                const double* x, bool new_x, double* cons)
+  {
+
+    assert(n==n_vars); assert(m==n_cons);
+    //for the base problem n_cons==2+n-3;
+
+    //local contributions to the constraints in cons are reset
+    for(auto j=0;j<m; j++) cons[j]=0.;
+
+    long long conidx{0};
+    //compute the constraint one by one.
+    // --- constraint 1 body --->  4*x_1 + 2*x_2 == 10
+    cons[conidx++] += 4*x[0] + 2*x[1];
+
+    // --- constraint 2 body ---> 2*x_1 + x_3
+    cons[conidx++] += 2*x[0] + 1*x[2];
+
+    // --- constraint 3 body --->   2*x_1 + 0.5*x_i, for i>=4
+    for(auto i=3; i<nx_; i++) {
+      cons[conidx++] += 2*x[0] + 0.5*x[i];
+    }
+
+    for(int i=0;i<S_;i++) {
+      for(int j=0;j<nx_-1;j++) {
+        cons[conidx+i*nx_+j] = x[nx_+i*nx_+j+1]-x[nx_+i*nx_+j];
+      }
+      cons[conidx+i*nx_+nx_-1] = (1-x[nx_+i*nx_]+xi_[i*nS_])*(1-x[nx_+i*nx_]+xi_[i*nS_]);
+      for(int j=1;j<nS_;j++) {
+        cons[conidx+i*nx_+nx_-1] += (x[nx_+i*nx_+j] + xi_[i*nS_+j])*(x[nx_+i*nx_+j] + xi_[i*nS_+j]);
+      }
+      for(int j=nS_;j<nx_;j++) {
+        cons[conidx+i*nx_+nx_-1] += x[nx_+i*nx_+j]*x[nx_+i*nx_+j];
+      }
+    }
+    
+    return true;
+  }
+  
+  bool eval_cons(const long long& n, const long long& m,
+                    const long long& num_cons, const long long* idx_cons,
+			              const double* x, bool new_x, double* cons)
+  {
+    return false;
+  }
+  bool eval_Jac_cons(const long long& n, const long long& m,
+                        const long long& num_cons, const long long* idx_cons,
+                        const double* x, bool new_x,
+                        const int& nnzJacS, int* iJacS, int* jJacS, double* MJacS)
+  {
+    return false;
+  }
+
+  bool eval_Jac_cons(const long long& n, const long long& m,
+                          const double* x, bool new_x,
+                          const int& nnzJacS, int* iJacS, int* jJacS, double* MJacS)
+  {
+    assert(n==n_vars); assert(m==n_cons);
+    assert(n>=3);
+    //2*(n-1) for basecase
+
+    assert(nnzJacS == 2*(nx_-1)+S_*(2*(nx_-1)+nx_));
+
+    int nnzit{0};
+    long long conidx{0};
+
+    if(iJacS!=NULL && jJacS!=NULL) {
+      // --- constraint 1 body --->  4*x_1 + 2*x_2 == 10
+      iJacS[nnzit] = conidx;   jJacS[nnzit++] = 0;
+      iJacS[nnzit] = conidx;   jJacS[nnzit++] = 1;
+      conidx++;
+
+      // --- constraint 2 body ---> 2*x_1 + x_3
+      iJacS[nnzit] = conidx;   jJacS[nnzit++] = 0;
+      iJacS[nnzit] = conidx;   jJacS[nnzit++] = 2;
+      conidx++;
+
+      // --- constraint 3 body --->   2*x_1 + 0.5*x_i, for i>=4
+      for(auto i=3; i<nx_; i++){
```

Comment:
missing space before bracket; the following lines are indented with 4 whitespaces instead of 2